app/controllers/concerns/foreman/controller/host_details.rb

```diff
@@ -45,12 +45,16 @@ def use_image_selected
   def assign_parameter(name, root = "")
     taxonomy_scope
     Taxonomy.as_taxonomy @organization, @location do
-      instance_variable_set("@#{name}",name.classify.constantize.where(:id => params["#{name}_id"]).first)
-      item = instance_variable_get("@#{controller_name.singularize}") || controller_name.classify.constantize.new(params[controller_name.singularize])
+      item = instance_variable_get("@#{controller_name.singularize}") || controller_name.classify.constantize.new(item_params)
+      instance_variable_set("@#{name}", item.send(name.to_sym))
       render :partial => root + name, :locals => { :item => item }
     end
   end
 
+  def item_params
+    send("#{controller_name.singularize}_params".to_sym)
```

Comment:
Oh yes it was part of your patch which I incorporated: https://github.com/timogoebel/foreman/commit/45acb33bbea1bd4a6fb02baed2b9bebe66ada170

Please clarify your comment so I can fix it, we are almost there.